index.js

```diff
@@ -1,18 +1,30 @@
 /* eslint-env node */
 'use strict';
+let objectHash = require('object-hash');
 
 module.exports = {
   name: 'ember-hbs-minifier',
 
-  setupPreprocessorRegistry(type, registry) {
-    if (type === 'parent') {
-      let HbsMinifierPlugin = require('./hbs-minifier-plugin');
+  _setupPreprocessorRegistry(app) {
+    let registry = app.registry;
+    let options = app.options || {};
+    let config = options['ember-hbs-minifier'] || {};
```

Comment:
I think this is the point at which we should apply the config defaults. If `config.skip` is not set we set it to `{ elements: ['pre'], components: ['no-minify'] }` by default.

Reply:
In that case, if the user provides a config, he need to pass the defaults (`pre` and `no-minify`) along with his config like below. Is it ok?

```javascript
{
  skip: {
    elements: ['pre', 'div'],
    components: ['no-minify', 'my-component']
  }
}
```

Reply:
yeah, I think that makes sense. otherwise it wouldn't be possible to *not* skip `<pre>`.